extensions/amp-smartlinks/0.1/amp-smartlinks.js

```diff
@@ -203,13 +203,22 @@ export class AmpSmartlinks extends AMP.BaseElement {
       'organization_type': 'publisher',
       'user': {
         'page_session_uuid': this.generateUUID_(),
-        'source_url': this.ampDoc_.getUrl(),
+        'source_url': this.getLocationHref_(),
         'previous_url': this.referrer_,
         'user_agent': this.ampDoc_.win.navigator.userAgent,
       },
     }));
   }
 
+  /**
+   * Retrieve url of the current doc.
+   * @return {string}
+   * @private
+   */
+  getLocationHref_() {
+    return window.location.href;
```

Comment:
Taking this data from the window/document is ok. However, we still need to use the correct `window` object. You should be able to use `this.win` instead of the `window` global. Could you pls try?

Reply:
@dvoytenko Thank you for the review! Just pushed up the changes!